pkg/abstract.go

```diff
@@ -183,26 +183,37 @@ func getMetricDataForQueries(
 	logger log.Logger) []cloudwatchData {
 	var getMetricDatas []cloudwatchData
 
+	mux := &sync.Mutex{}
+	var wg sync.WaitGroup
+
 	// For every metric of the job
 	for _, metric := range discoveryJob.Metrics {
-		// Get the full list of metrics
-		// This includes, for this metric the possible combinations
-		// of dimensions and value of dimensions with data
-		tagSemaphore <- struct{}{}
+		wg.Add(1)
+		go func(m *Metric) {
+			defer wg.Done()
+			// Get the full list of metrics
+			// This includes, for this metric the possible combinations
+			// of dimensions and value of dimensions with data
+			metricsList, err := getFullMetricsList(ctx, svc.Namespace, m, clientCloudwatch)
+
+			if err != nil {
+				level.Error(logger).Log("msg", "Failed to get full metric list", "err", err, "metric_name", m.Name, "namespace", svc.Namespace)
```

Comment:
Maybe we could add a prom metrics to track both this errors and the scenario below with zero resources?